source/kernel/io_trace.c

```diff
@@ -122,12 +122,18 @@ static void bio_complete_event(void *ignore,
                                struct request_queue *q,
                                struct bio *bio,
                                int error) {
-    uint32_t dev_id;
+    struct gendisk *disk;
+    uint64_t dev_id;
     unsigned cpu = get_cpu();
     struct iotrace_context *iotrace = iotrace_get_context();
 
     if (iotrace_bdev_is_added(&iotrace->bdev, cpu, q)) {
-        dev_id = disk_devt(IOTRACE_BIO_GET_DEV(bio));
+        disk = IOTRACE_BIO_GET_DEV(bio);
+        if (disk) {
+            dev_id = disk_devt(disk);
+        } else {
+            dev_id = (uint64_t) q->dev;
```

Comment:
The trace could also be dropped, similar to what's done in RH:
https://www.redhat.com/archives/dm-devel/2013-January/msg00144.html